x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/SequenceMatcher.java

```diff
@@ -72,11 +73,16 @@ public void clear() {
     // Set of completed sequences - separate to avoid polluting the other stages
     // It is a set since matches are ordered at insertion time based on the ordinal of the first entry
     private final Set<Sequence> completed;
+    private final Set<Sequence> toCheckForMissing;
     private final long maxSpanInNanos;
 
     private final boolean descending;
 
     private final Limit limit;
+    private final boolean[] missingEvents;
+    protected final int firstPositiveStage;
```

Comment:
The "positive" and "negative" meaning needs to reflect in the documentation as well, since we have error messages that refer to them.

Reply:
I double-checked the error messages and everything seems OK.
We'll have to update the docs when this PR is final.